File: api-frontend/apps/web-antd/src/api/system/city.ts
import { requestClient } from '#/api/request';

export interface CityRecord {
  id: number;
  pid: number;
  level: number;
  name: string;
  citycode: string;
  adcode: string;
  p_adcode: string;
  lng: number;
  lat: number;
  sort: number;
  created_at: string;
  children?: CityRecord[];
}

export function getCityList(pid: number = 0) {
  return requestClient.get<CityRecord[]>('/cities', { params: { pid } });
}

export function getCityTree(maxLevel: number = 2) {
  return requestClient.get<CityRecord[]>('/cities/tree', { params: { max_level: maxLevel } });
}

export function createCity(data: Partial<CityRecord>) {
  return requestClient.post('/cities', data);
}

export function updateCity(id: number, data: Partial<CityRecord>) {
  return requestClient.put(`/cities/${id}`, data);
}

export function deleteCity(id: number) {
  return requestClient.delete(`/cities/${id}`);
}

File: api-backend/internal/model/city.go
package model

type City struct {
	BaseModel
	PID      uint    `gorm:"column:pid;default:0;index" json:"pid"`
	Level    int8    `gorm:"default:0" json:"level"`
	Name     string  `gorm:"size:100;not null" json:"name"`
	Citycode string  `gorm:"size:20" json:"citycode"`
	Adcode   string  `gorm:"size:20" json:"adcode"`
	PAdcode  string  `gorm:"size:20" json:"p_adcode"`
	Lng      float64 `gorm:"type:decimal(10,6)" json:"lng"`
	Lat      float64 `gorm:"type:decimal(10,6)" json:"lat"`
	Sort     int     `gorm:"default:0" json:"sort"`
}

func (City) TableName() string {
	return "cities"
}
